io/io/src/RRawFileCreate.cxx

```diff
@@ -0,0 +1,44 @@
+// @(#)root/io:$Id$
+// Author: Jakob Blomer
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RConfig.h>
+#include <ROOT/RRawFile.hxx>
+#ifdef _WIN32
+#include <ROOT/RRawFileWin.hxx>
+#else
+#include <ROOT/RRawFileUnix.hxx>
+#endif
+
+#include "TPluginManager.h"
+#include "TROOT.h"
+
+std::unique_ptr<ROOT::Internal::RRawFile>
+ROOT::Internal::RRawFile::Create(std::string_view url, ROptions options)
```

Comment:
Isn't it a problem that this implementation is in RIO and the rest of the declaration/implementation is in ROOTFoundationLIte?

Reply:
As discussed, `RRawFile::Create()` could work with local files only in RIOLite.  For other transports (e.g. "http"), `RRawFile::Create` can call a static function pointer (if set) of `RRawFile`.  This function pointer would be set when `TROOT` is created.

Reply:
@pcanal I am a bit confused about where I need to set a static function pointer, could you please take a look at the comment above of @jblomer. I think I can't find a place where `TROOT` is created, can you suggest please where to look?

Reply:
@oshadura The file RRawFileCreate.cxx would need to move to RIOLite.  The part of the code
```
if (transport == "http" || transport == "https") {
...
}
```
would move to a function in RIO (since it make more sense than being in TROOT.cxx .. but it technically could be there too).

Then via a global static function initialization that file in RIO would register the function with RRawFile.  Similar to
```
static int gRegisterTrigger = ROOT::Internal::RRawFile::RegisterCreateHttpFunction( & CreateHttpRRawFile );
```
with `CreateHttpRRawFile` containing the code extract from `RRawFile::Create`

Reply:
We could also be even more helpful and offer an abstract interface, with multiple of these protocol handlers able to register themselves with foundationiolite. That might be both easier and more useful? E.g. what about Davix support (which might be in a dedicated library because it depends on libXrdClient)?